gwpy/segments/tests/test_segments.py

```diff
@@ -82,6 +84,15 @@ def test_coalesce(self):
         assert_segmentlist_equal(c, [(1, 2), (3, 5)])
         assert isinstance(c[0], self.ENTRY_CLASS)
 
+    def test_to_table(self, segmentlist):
+        segtable = segmentlist.to_table()
+        assert isinstance(segtable, Table)
+        assert_array_equal(segtable['index'], list(range(len(segmentlist))))
+        assert_array_equal(segtable['start'], [seg[0] for seg in segmentlist])
+        assert_array_equal(segtable['end'], [seg[1] for seg in segmentlist])
+        assert_array_equal(
+            segtable['duration'], [abs(seg) for seg in segmentlist])
+
```

Comment:
Can you just instead use `assert_table_equal` and build the expected table on the fly? Something like:

```python
    def test_to_table(self, segmentlist):
        segtable = segmentlist.to_table()
        assert_table_equal(
            segtable,
            Table(
                rows=[
                    (0, 1, 2, 1),
                    (1, 3, 4, 1),
                    (2, 4, 6, 2),
                    (3, 8, 10, 2),
                ],
                names=('index', 'start', 'end', 'duration'),
            ),
        )
```
(untested)

Reply:
@duncanmmacleod, your proposed unit test passes (and runs about 6.5 times faster), so I've implemented it in the latest commit.